src/dbnode/storage/index.go

```diff
@@ -1162,6 +1263,21 @@ func (i *nsIndex) flushBlockSegment(
 	return preparedPersist.Persist(builder)
 }
 
+func (i *nsIndex) allowDuplicatesWriteError(err error) error {
```

Comment:
nit: `sanitizeError` maybe?

Reply:
Hm so that has a little less information I think and doesn't really say what it's doing, so you have to go read the method. That's why I opted for a less generic name here that says what it does vs a more generic "sanitizeFoo". But yeah, I'd lean towards more information in the name so you don't have to necessarily visit function to know what it's doing. 

Could go with `sanitizeAllowDuplicatesWriteError(...)`? I'll make that change.